backend/apps/bookings/models.py

```diff
@@ -49,9 +58,50 @@ def cancel_booking(self):
         self.save()
         return True
 
-    def booking_end_time(self, duration_in_minutes=60):
-        return self.session_date + timedelta(minutes=duration_in_minutes)
+    def booking_duration(self):
+        return self.session_end_time - self.session_date
 
     @classmethod
     def get_bookings_by_status(cls, status):
         return cls.objects.filter(booking_status=status)
+
+    def is_expired(self):
+        return self.session_end_time < now()
+
+    def clean(self):
+        if self.session_end_time <= self.session_date:
+            raise ValidationError("Session end time must be after the session start time.")
+
+    def save(self, *args, **kwargs):
+        """Run full validation before saving."""
+        self.full_clean()  # Ensures validation runs before saving
+        super().save(*args, **kwargs)
+
+
+class Review(models.Model):
+    tutor = models.ForeignKey(User, on_delete=models.CASCADE, related_name="reviews")
+    reviewer = models.ForeignKey(User, on_delete=models.CASCADE, related_name="reviewer")
+    rating = models.PositiveSmallIntegerField(validators=[MinValueValidator(1), MaxValueValidator(5)])
+    feedback = models.TextField(null=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    is_visible = models.BooleanField(default=True)  # Whether the review is visible to others
+    is_moderated = models.BooleanField(
+        default=False)  # Whether the review has been moderated (e.g., checked for inappropriate content)
+
+    def __str__(self):
+        return f"Review for {self.tutor.username} by {self.reviewer.username}"
+
+    class Meta:
+        ordering = ['-created_at']  # Sort reviews by most recent first
```

Comment:
```suggestion
    class Meta:
        ordering = ['-created_at']  # Sort reviews by most recent first
        unique_together = ('tutor', 'reviewer') # Review constraints
```
Review constraints, ensures that a student can only leave one review per booking.